This patch adds the capability to delete email from the new email notification bar.
Index: /usr/palm/applications/com.palm.app.email/app/controllers/email-dashboard-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/email-dashboard-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/email-dashboard-assistant.js
@@ -32,9 +32,17 @@ var EmailDashboardAssistant = Class.crea
 	tapHandler: function(event) {
 		// Ignore additional taps in case the user is impatient
 		var target = event.target;
+
 		if (!target.hasClassName('email-dashboard-tap-target')) {
 			target = target.up('div.email-dashboard-tap-target');
 		}
+	
+		var tDelete = false;
+			
+		//check to see if the trash icon was tapped
+		if ( target.hasClassName('email-dashboard-tap-target-delete')) {
+			tDelete = true;
+		}
 
 		if (target) {
 			var accountId = target.readAttribute('accountid');
@@ -42,7 +50,15 @@ var EmailDashboardAssistant = Class.crea
 			if (accountId == "single") {
 				var folderId = target.readAttribute('folderid');
 				var messageId = target.readAttribute('messageid');
-				this.showSingleMessage(accountId, folderId, messageId);
+	
+				if(tDelete)
+				{
+					this.deleteSingleMessage(accountId, folderId, messageId);
+				}
+				else
+				{
+					this.showSingleMessage(accountId, folderId, messageId);
+				}
 			} else if (accountId == "all") {
 				this.showAccount(); // just display the account scene
 			} else {
@@ -145,6 +161,24 @@ var EmailDashboardAssistant = Class.crea
 	                });
 	},
 
+	 
+	deleteSuccess: function(folderId) {
+		//do nothing but leaving this for future use.
+	},
+
+	deleteFailure: function(messageId) {
+		this.notificationAssistant.clear(undefined, undefined, messageId);
+	},
+	
+	deleteSingleMessage: function(account, folder, message) {
+		var deleteReq = new Mojo.Service.Request('palm://com.palm.mail', {
+			method: 'setDeleted',
+			parameters: {'message':message, 'value': true },
+			onSuccess: this.deleteSuccess.bind(this, folder),
+			onFailure: this.deleteFailure.bind(this, message)
+			});
+	},
+	
 	showFolder: function(accountId, folderId, resp) {
 		var folderInfo = { folderId: folderId,
 		                   accountId: accountId,
Index: /usr/palm/applications/com.palm.app.email/app/views/email-dashboard/multiple-accounts-template.html
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/views/email-dashboard/multiple-accounts-template.html
+++ /usr/palm/applications/com.palm.app.email/app/views/email-dashboard/multiple-accounts-template.html
@@ -5,7 +5,11 @@
     </div>
      <div class="palm-dashboard-icon email">
      </div>
-  </div>   
+  </div> 
+  <div class="palm-dashboard-icon-container email-dashboard-tap-target email-dashboard-tap-target-delete"  accountid="single" folderid="#{folder}" messageid="#{message}">
+	<div class="palm-dashboard-icon delete">
+	</div>
+  </div>
   <div class="palm-dashboard-text-container email-dashboard-tap-target" accountid="single" folderid="#{folder}" messageid="#{message}">
      <div class="dashboard-title">
       #{from}
Index: /usr/palm/applications/com.palm.app.email/app/views/email-dashboard/single-account-template.html
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/views/email-dashboard/single-account-template.html
+++ /usr/palm/applications/com.palm.app.email/app/views/email-dashboard/single-account-template.html
@@ -5,6 +5,10 @@
     </div>
      <div class="palm-dashboard-icon #{accountDomain}">
      </div>
+  </div> 
+  <div class="palm-dashboard-icon-container email-dashboard-tap-target email-dashboard-tap-target-delete"  accountid="single" folderid="#{folder}" messageid="#{message}">
+	<div class="palm-dashboard-icon delete">
+        </div>
   </div>   
   <div class="palm-dashboard-text-container email-dashboard-tap-target" accountid="single" folderid="#{folder}" messageid="#{message}">
      <div class="dashboard-title">
Index: /usr/palm/applications/com.palm.app.email/app/views/email-dashboard/single-email-template.html
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/views/email-dashboard/single-email-template.html
+++ /usr/palm/applications/com.palm.app.email/app/views/email-dashboard/single-email-template.html
@@ -1,8 +1,12 @@
-<div class="dashboard-notification-module single">
+<div class="dashboard-notification-module">
   <div class="palm-dashboard-icon-container email-dashboard-tap-target" accountid="#{account}">
      <div class="palm-dashboard-icon #{accountDomain}">
      </div>
   </div>   
+  <div class="palm-dashboard-icon-container email-dashboard-tap-target email-dashboard-tap-target-delete"  accountid="single" folderid="#{folder}" messageid="#{message}">
+	<div class="palm-dashboard-icon delete">
+	</div>
+  </div>
   <div class="palm-dashboard-text-container email-dashboard-tap-target" accountid="single" folderid="#{folder}" messageid="#{message}">
      <div class="dashboard-title">
       #{from}
Index: /usr/palm/applications/com.palm.app.email/stylesheets/notifications.css
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/stylesheets/notifications.css
+++ /usr/palm/applications/com.palm.app.email/stylesheets/notifications.css
@@ -78,3 +78,10 @@ Copyright 2009 Palm, Inc.  All rights re
 .palm-dashboard-icon.gmail			{ background: url(../images/notification-large-gmail.png) center left no-repeat; }
 .palm-dashboard-icon.yahoo			{ background: url(../images/notification-large-yahoo.png) center left no-repeat; }
 .palm-dashboard-icon.aol			{ background: url(../images/notification-large-aol.png) center left no-repeat; }
+
+/*To support delete of email from notification*/
+.palm-dashboard-icon.delete                        { background: url(../images/icon_trash.png) center center no-repeat; }
+
+.email-dashboard-tap-target-delete {
+}
+
